tools/build-addons.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+// Usage: e.g. node build-addons.js <path to node-gyp> <directory>
+
+const child_process = require('child_process');
+const path = require('path');
+const fs = require('fs').promises;
+const util = require('util');
+
+const execFile = util.promisify(child_process.execFile);
+
+const parallelization = +process.env.JOBS || require('os').cpus().length;
+const nodeGyp = process.argv[2];
+
+async function runner(directoryQueue) {
+  if (directoryQueue.length === 0)
+    return;
+
+  const dir = directoryQueue.shift();
+  const next = () => runner(directoryQueue);
+
+  try {
+    // Only run for directories that have a `binding.gyp`.
+    // (https://github.com/nodejs/node/issues/14843)
+    await fs.stat(path.join(dir, 'binding.gyp'));
+  } catch (err) {
+    if (err.code === 'ENOENT' || err.code === 'ENOTDIR')
+      return next();
+    throw err;
+  }
+
+  console.log(`Building addon in ${dir}`);
+  const { stdout, stderr } =
+    await execFile(process.execPath, [nodeGyp, 'rebuild', `--directory=${dir}`],
+                   {
+                     stdio: 'inherit',
```

Comment:
Do we need this for `execFile`? We will write the output to this process later anyway?

Reply:
@joyeecheung This is so we don’t get interleaved output from multiple builds running in parallel … will add a comment :)